### PR TITLE
Fixed rtti for *nix

### PR DESCRIPTION
**Pull request by gaaclarke**
This is required for building vulkan validation layers.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: build/config/compiler/BUILD.gn [hunk 1]
```diff
@@ -761,6 +761,8 @@ config("no_chromium_code") {
 config("rtti") {
   if (is_win) {
     cflags_cc = [ "/GR" ]
+  } else {
+    cflags_cc = [ "-frtti" ]
```
Comment (dnfield) [inline build/config/compiler/BUILD.gn:765]: Below for no-rtti it gets applied to both cflags_cc and cflags_objcc. Should we do the same here?

Comment (dnfield) [reply build/config/compiler/BUILD.gn]: Yeah, we should just do that here too so someone using this config doesn't get surprised when it doesn't work as expected on ObjC++

**Review by dnfield — APPROVED** [review]

**Comment by zanderso** [comment]
It looks like this gets applied to libcxx, and libcxxabi:

https://github.com/flutter/buildroot/blob/master/build/secondary/third_party/libcxx/BUILD.gn#L83
https://github.com/flutter/buildroot/blob/master/build/secondary/third_party/libcxxabi/BUILD.gn#L61

and icu in its BUILD.gn from third_party. Wondering if this will have any wider impact on binary size or performance.

**Comment by dnfield** [comment]
My understanding of this change it that it should either fix a compile time error or be a no-op. 

If a TU was using RTTI and the compiler needed this flag but wasn't getting it, that TU wouldn't build before. If the compiler didn't need the flag explicitly, then it already was building that TU with RTTI.

@gaaclarke are you sure that the TU you're building locally isn't getting both `-fno-rtti` and now `-frtti` and this is just changing the order? i.e. are you forgetting to do something like `configs -= no_rtti` somewhere?

**Comment by zanderso** [comment]
> My understanding of this change it that it should either fix a compile time error or be a no-op.

For the deps I mentioned above, they were previously getting neither `-frtti` nor `-fno-rtti`, and after this change will get `-frtti`.

**Comment by dnfield** [comment]
Correct - but AFAIK `-frtti` is the default for Clang. That's making me wonder if The problem @gaaclarke is seeing is that `-fno-rtti` is still getting applied, but this change makes the command line have _both_ `-fno-rtti` _and_ `-frtti` in whatever order makes clang end up ignoring the `-fno-rtti`.

**Comment by dnfield** [comment]
@gaaclarke can you confirm this? Maybe we should revert this if my suspicion is correct now - we shouldn't rely on what clang does when you specify both `-fno-rtti` and `-frtti` in the same invocation.

**Comment by gaaclarke** [comment]
@dnfield you're right, it is getting both flags.

```
[1/2] /Users/aaclarke/dev/engine/src/buildtools/mac-x64/goma/gomacc  ../../buildtools/mac-x64/clang/bin/clang++ -MD -MF obj/third_party/vulkan-deps/vulkan-validation-layers/src/layers/libVkLayer_khronos_validation.layer_options.o.d -DUSE_OPENSSL=1 -D_LIBCPP_DISABLE_AVAILABILITY=1 -D_LIBCPP_DISABLE_VISIBILITY_ANNOTATIONS -D_LIBCPP_ENABLE_THREAD_SAFETY_ANNOTATIONS -D_DEBUG -DVULKAN_NON_CMAKE_BUILD -DVK_ENABLE_BETA_EXTENSIONS -DVK_USE_PLATFORM_METAL_EXT -I../.. -Igen -I../../third_party/libcxx/include -I../../third_party/libcxxabi/include -I../../build/secondary/third_party/libcxx/config -I../../third_party/vulkan-deps/vulkan-validation-layers/src/layers -I../../third_party/vulkan-deps/vulkan-validation-layers/src/layers/external -I../../third_party/vulkan-deps/vulkan-validation-layers/src/layers/generated -I../../third_party/vulkan-deps/vulkan-headers/src/include -I../../third_party/vulkan-deps/spirv-tools/src -I../../third_party/vulkan-deps/spirv-tools/src/include -I../../third_party/vulkan-deps/spirv-headers/src/include -isysroot /Users/aaclarke/dev/engine/src/out/host_debug_unopt/gen/SDKs/MacOSX13.0.sdk -mmacosx-version-min=10.14.0 -fno-strict-aliasing -fstack-protector-all -arch x86_64 -fcolor-diagnostics -fvisibility=hidden -Wstring-conversion -Wnewline-eof -O0 -g2 -Wno-deprecated-non-prototype -Wno-newline-eof -Wno-typedef-redefinition -Wendif-labels -Werror -Wno-missing-field-initializers -Wno-unused-parameter -Wno-unused-but-set-parameter -Wno-unused-but-set-variable -Wno-implicit-int-float-conversion -Wno-c99-designator -Wno-deprecated-copy -Wno-psabi -Wno-unqualified-std-cast-call -Wno-non-c-typedef-for-linkage -Wno-range-loop-construct -Wunguarded-availability -Wno-deprecated-declarations -Wno-conversion -Wno-deprecated-copy -Wno-extra-semi -Wno-implicit-fallthrough -Wno-missing-field-initializers -Wno-newline-eof -Wno-sign-compare -Wno-unused-const-variable -Wno-extra-semi -Wno-unused-function -Wno-unguarded-availability-new -Wno-extra-semi -Wno-redundant-parens -fvisibility-inlines-hidden -std=c++17 -fno-rtti -nostdinc++ -nostdinc++ -fvisibility=hidden -fno-exceptions -stdlib=libc++ -Wno-newline-eof -frtti -std=c++17 -Wno-nullability-completeness  -c ../../third_party/vulkan-deps/vulkan-validation-layers/src/layers/layer_options.cpp -o obj/third_party/vulkan-deps/vulkan-validation-layers/src/layers/libVkLayer_khronos_validation.layer_options.o
```

Now the question is, where is "fno-rtti" coming from since it's not coming from `"//build/config/compiler:no_rtti"`.

**Comment by gaaclarke** [comment]
Here's the configs for the target, none of them mention rtti, so it's probably hidden somewhere in a cflags somewhere:
```
//build/config:feature_flags
//build/config/compiler:compiler
//build/config/compiler:cxx_version_default
//build/config/compiler:compiler_arm_fpu
//build/config/compiler:chromium_code
//build/config/compiler:default_include_dirs
//build/config/compiler:runtime_library
//third_party/libcxxabi:libcxxabi_config
//third_party/libcxx:libcxx_config
//build/config/gcc:no_exceptions
//build/config/gcc:symbol_visibility_hidden
//build/config:symbol_visibility_hidden
//build/config/mac:sdk
//build/config/clang:extra_warnings
//build/config/clang:find_bad_constructs
//build/config:debug
//build/config/compiler:no_optimize
//build/config/compiler:default_optimization
//build/config/compiler:symbols
//build/config:default_libs
//build/config/mac:mac_dynamic_flags
//build/config/mac:mac_executable_flags
//build/config/compiler:rtti
//flutter/impeller:impeller_public_config
//third_party/skia:fontmgr_mac_ct_public
//third_party/skia:gpu_public
//third_party/skia:gpu_shared_public
//third_party/skia:jpeg_encode_public
//third_party/skia:png_encode_public
//third_party/skia:webp_encode_public
//third_party/skia:fontmgr_mac_ct_factory_public
//third_party/skia:jpeg_decode_public
//third_party/skia:png_decode_public
//third_party/skia:webp_decode_public
//third_party/skia:wuffs_public
//flutter/impeller/fixtures:embed_embed_mtl_shader_fixtures
//flutter/impeller/fixtures:reflect_reflect_mtl_shader_fixtures
//flutter/impeller/fixtures:embed_embed_vk_shader_fixtures
//flutter/impeller/entity:embed_embed_mtl_entity_shaders
//flutter/impeller/entity:reflect_reflect_mtl_entity_shaders
//flutter/impeller/entity:embed_embed_vk_entity_shaders
//flutter/impeller/entity:embed_embed_mtl_framebuffer_blend_entity_shaders
//flutter/impeller/entity:reflect_reflect_mtl_framebuffer_blend_entity_shaders
//flutter/impeller/entity:embed_embed_vk_framebuffer_blend_entity_shaders
//flutter/impeller/entity:embed_embed_mtl_modern_entity_shaders
//flutter/impeller/entity:reflect_reflect_mtl_modern_entity_shaders
//flutter/impeller/entity:embed_embed_vk_modern_entity_shaders
//flutter/impeller/runtime_stage:runtime_stage_config
//third_party/flatbuffers:flatbuffers_public_configs
//flutter:config
//flutter/common:flutter_config
//flutter/impeller/renderer:embed_embed_mtl_compute_shaders
//flutter/impeller/renderer:reflect_reflect_mtl_compute_shaders
//flutter/impeller/renderer:embed_embed_vk_compute_shaders
//flutter/impeller/scene/shaders:embed_embed_mtl_shaders
//flutter/impeller/scene/shaders:reflect_reflect_mtl_shaders
//flutter/impeller/scene/shaders:embed_embed_vk_shaders
//flutter/impeller/playground/imgui:embed_embed_mtl_imgui_shaders
//flutter/impeller/playground/imgui:reflect_reflect_mtl_imgui_shaders
//flutter/impeller/playground/imgui:embed_embed_vk_imgui_shaders
//third_party/glfw:relative_glfw_headers
//third_party/imgui:imgui_headers
```

**Comment by dnfield** [comment]
The rtti config might not get listed there because it's added as a default. I'm not sure how GN handles those when listing configs.

**Pull request merged**